include/zephyr/toolchain/gcc.h

```diff
@@ -209,8 +209,17 @@ do {                                                                    \
 #define __ramfunc
 #elif defined(CONFIG_ARCH_HAS_RAMFUNC_SUPPORT)
 #if defined(CONFIG_ARM)
+#if defined(__clang__)
```

Comment:
I was about asking why this is not handled in https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/toolchain/llvm.h, as that header is LLVM specific, but I noticed there are more clang related safeguards in gcc.h.

Probably should have a bigger overhaul, but as that would be outside scope of this PR, then proposed changes are accepted as it is aligned with existing code.

Reply:
Yeah, my expectation was also that there would be some common file which would get included by _llvm.h_ and _gcc.h_.